### PR TITLE
Emc/remove next proxies

### PR DESCRIPTION
**Pull request by 0xemc**
## Description

* Updates API url references to use constants
* Adds CORS middleware to fastify
* Removes proxy NextJS routes
* Updates Carbonmark to point to localhost in development (we need a better solution here)

## Remaining
- [x] Configure start scripts to start api when developing carbonmark
- [x] Confirm that we want preview API deployments to be accessible via localhost

### CODE REVIEW

**Comment by 0xemc** [comment]
@Atmosfearful @ladytrekker I'm wondering if we remove these proxies and we want to test locally against the preview or production endpoints we might have a problem (i.e we'd need to enable the CORS middleware in those environments 😬)

Thoughts?

**Review by Atmosfearful — APPROVED** [review]

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
Whoops meant to click `request changes`. Lets get the host port sorted b4 merging

File: carbonmark-api/plugins/cors.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const fp = require("fastify-plugin");
+const cors = require("@fastify/cors");
+
+const ALLOWED_ORIGINS = [
+  "localhost",
+  "carbonmark.com",
```
[the remark this answers is not in the document]
Comment (0xemc) [reply carbonmark-api/plugins/cors.js]: Yea I was wondering that too myself. I've just set it open now and set a rate-limit of 100 request per minute. If you want to change that let me know

File: carbonmark/lib/constants.ts [hunk 2]
```diff
@@ -48,33 +57,31 @@ export const config = {
     baseUrl: {
       production: "https://www.carbonmark.com",
       preview: "https://" + process.env.NEXT_PUBLIC_VERCEL_URL, // note: won't take custom subdomains like staging.carbonmark.com, takes the vercel generated url instead
-      development: "http://localhost:3002",
+      development: "http://localhost:3000",
```
[the remark this answers is not in the document]
Comment (0xemc) [reply carbonmark/lib/constants.ts:60]: That makes sense, updated to 3002

File: carbonmark/lib/constants.ts [hunk 3]
```diff
@@ -48,33 +57,31 @@ export const config = {
...
+      production: polygonNetworks.mainnet.blockExplorerUrls[0],
+      preview: polygonNetworks.testnet.blockExplorerUrls[0],
+      development: polygonNetworks.testnet.blockExplorerUrls[0],
     },
     api: {
-      mainnet: "https://api.carbonmark.com/api",
-      testnet: "https://api.carbonmark.com/api",
+      production: "https://api.carbonmark.com/api",
+      preview: "https://staging-api.carbonmark.com/api",
+      development: "http://localhost:3000/api",
```
[the remark this answers is not in the document]
Comment (0xemc) [reply carbonmark/lib/constants.ts]: I agree with all of that, updated to:

* use `3003` as the dev port
* default to the staging url in development
* allow the developer to override the url via NEXT_PUBLIC_CARBONMARK_API_URL env if necessary

File: carbonmark/hooks/useFetchProject.ts [hunk 4]
```diff
@@ -7,8 +8,9 @@ export const useFetchProject = (
   projectID: string,
   options?: SWRConfiguration
 ) => {
+  console.log("API.PROHJECTS", urls.api.projects);
```
Comment (Atmosfearful) [inline carbonmark/hooks/useFetchProject.ts:11]: left a debug in here

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
Something funky with the commit history, see if `git pull -r origin staging && git push -f` fixes it?

Hopefully that removes the i18n and css changes im seeing in the diff

Otherwise LGTM!

File: carbonmark/hooks/useFetchProject.ts (hunk 4 above)
Comment (0xemc) [reply carbonmark/hooks/useFetchProject.ts]: I'm glad it wasn't one of my "wtf" logs..

Removed

**Comment by 0xemc** [comment]
> Something funky with the commit history, see if `git pull -r origin staging && git push -f` fixes it?
> 
> Hopefully that removes the i18n and css changes im seeing in the diff
> 
> Otherwise LGTM!

The diff was broken because I configured the PR to target a different destination branch (the fix broken build one) but now that's been merged in. Updated now

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**